adapters/x509_openssl.c

```diff
@@ -219,6 +219,7 @@ int x509_openssl_add_pem_file_key(SSL_CTX* ssl_ctx, const char* x509privatekey)
     return result;
 }
 
+#ifndef OPENSSL_NO_ENGINE
```

Comment:
I believe the error is now caused by this file. 
I would do a search for `#include "openssl/engine.h"` and add the `#ifdef` everywhere to make the older OpenSSL version compile.